gcc/9/patches/0032-gcc9-Don-t-define-__STRICT_ANSI__.-Doing-so.patch

```diff
@@ -0,0 +1,38 @@
+From 3dfc96d88a8b2336e528864452490e163158d35f Mon Sep 17 00:00:00 2001
+From: "ola.soder@axis.com" <ola.soder@axis.com>
+Date: Wed, 27 Jan 2021 18:56:32 +0100
+Subject: [PATCH 32/32] gcc9: Don't define __STRICT_ANSI__. Doing so hides C99
+ features when configuring libstdc++.
+
+---
+ libstdc++-v3/config/os/amigaos/os_defines.h | 11 +++++++++--
+ 1 file changed, 9 insertions(+), 2 deletions(-)
+
+diff --git a/libstdc++-v3/config/os/amigaos/os_defines.h b/libstdc++-v3/config/os/amigaos/os_defines.h
+index 6b67630b7be7102a9dfb7c104deac6293a13c017..3f2d894b6c28e493e38c114da8d4f8528c4e3a12 100644
+--- a/libstdc++-v3/config/os/amigaos/os_defines.h
++++ b/libstdc++-v3/config/os/amigaos/os_defines.h
+@@ -36,11 +36,18 @@
+ // No ioctl() on AmigaOS
+ #define _GLIBCXX_NO_IOCTL 1
+ 
+ #ifdef __NEWLIB__
+ #define _GLIBCXX_USE_C99_STDINT_TR1 1
+ #define _GLIBCXX_USE_C99 1
+-/* Temporary until newlib behaves properly */
+-#undef __STRICT_ANSI__
```

Comment:
Sorry, I found out that I did not submit the review. Not sure if you saw it.

Reply:
I'll fix that. I wasn't sure how much 'paper trail' to produce. When looking at the original patch (0011) my guess is that we can skip it completely, the new newlib (!?)  implements strtold and so does the proposed clib2.

Reply:
I forgot; I think newlib behaves the way it should. It hides everything >C89 when __STRICT_ANSI__ is set, just like glibc The question is why __STRICT_ANSI__  is set during configuration. Either way, just like you say, the undef is temporary, but I don't think newer versions of newlib will solve the problem.

Reply:
IIRC ```__STRICT_ANSI__``` "just" hides any gnu extensions of the C standard at the currently selected C level in the sense of the synax of the language. I'm not sure if shall have implications of the defined functions or at least it should not be the only parameter. This probably was a workaround because newlib defined some function in sope of ```__STRICT_ANSI__``` that were present in c11, but c11 implies ```__STRICT_ANSI__``` and so configure did not produce the desired results. That does not mean that the behaviour is the expected one but this is how I observed it.

Reply:
> I'll fix that. I wasn't sure how much 'paper trail' to produce. When looking at the original patch (0011) my guess is that we can skip it completely, the new newlib (!?) implements strtold and so does the proposed clib2.

At least for newlib, we have to support the older version for quite a while until new newlib and SDK is circulated.

Reply:
>IIRC __STRICT_ANSI__ "just" hides any gnu extensions of the C standard at the currently selected C level in the sense of the synax of the language. I'm not sure if shall have implications of the defined functions or at least it should not be the only parameter.

I don't think it's just about extensions. From the gcc documentation:

\_\_STRICT_ANSI__
    GCC defines this macro if and only if the -ansi switch, or a -std switch specifying strict conformance to some version of ISO C, was specified when GCC was invoked. It is defined to `1'. This macro exists primarily to direct GNU libc's header files to restrict their definitions to the minimal set found in the 1989 C standard. 

There's always something hidden in there to cause confusion, 'primarily', what other use cases are there? :-)

> At least for newlib, we have to support the older version for quite a while until new newlib and SDK is circulated.

Yes, bits and pieces from all over the place is not a good idea. I really don't understand why they don't open up newlib.

Reply:
> > IIRC **STRICT_ANSI** "just" hides any gnu extensions of the C standard at the currently selected C level in the sense of the synax of the language. I'm not sure if shall have implications of the defined functions or at least it should not be the only parameter.
> 
> I don't think it's just about extensions. From the gcc documentation:
> 
> __STRICT_ANSI__
> GCC defines this macro if and only if the -ansi switch, or a -std switch specifying strict conformance to some version of ISO C, was specified when GCC was invoked. It is defined to `1'. This macro exists primarily to direct GNU libc's header files to restrict their definitions to the minimal set found in the 1989 C standard.
> 
> There's always something hidden in there to cause confusion, 'primarily', what other use cases are there? :-)

Yes, I'm aware of this but following my observation I ignored it. I always had the feeling this was old documentation (especially the last part, the first part IMO holds true) because it does not reflect what gcc does at the moment. My bet was is that the usage of the macro was extended and the last part was just an example. One would have too look at the history of gcc and the gnu libc to see when this confusion started. My observation was based on:

```
 $ gcc -std=c++17 -x c++ -dM -E - < /dev/null | grep STRICT
```
or

```
 $ gcc -std=c11 -dM -E - < /dev/null | grep STRICT
```

in contrast to

```
 $ gcc -std=gnu++17 -x c++ -dM -E - < /dev/null | grep STRICT
```
or

```
 $ gcc -std=gnu11 -dM -E - < /dev/null | grep STRICT
```
(on am amd64 glibc-based linux system).

Note that apparently ```__STRICT_ANSI___``` is also defined in C++. All in all I believe that the real fix (if you want to call it that way) should be done in the respective clib files, but temporarily it cal be addressed via patches.

Reply:
Interesting, thanks for the info.

Do you have any idea why newlib doesn't check __STDC_VERSION__ instead of __STRICT_ANSI__?

Reply:
I fixed the __STRICT_ANSI__ thing and bumped clib2 while I was at it. It contains a couple of fixes and some improvements in C99 compatibility, tgmath, iso646 and partial wchar support.